File: Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
       int x=0,c;
        for(int n:nums){
            if(n==x){
                c++;
            }
            else{
                c--;
            }
            if(c<=0){
                x = n;
                c = 1;
            }
        }
        return x;
    }
};

File: Leetcode/Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        x,c = 0,0
        for n in nums:
            if n==x:
                c+=1
            else:
                c-=1
            if c<=0:
                x = n
                c = 1
        return x